### ISSUE TITLE
[Share plugin - Permalink] Add a permalink functionnality

### ISSUE THREAD

**Issue by catmorales** [georchestra/mapstore2-georchestra#583]
## Description
'*Permalink*  shares current user session (custom layers in the TOC, custom background layers, all user customizations) and not only the initial map of the Context. It should be public by default.
 If some layers in it are protected by a geoserver rule, the layer will be ommited from the permalink, the functionnality should advice the user of it. 
Permalink is necessarity to complete the "share" plugin. His behaviour should be similar as Mapfishapp's permalink.

The Mapstore2 "share " plugin shares only the original context not the current context customized by user.
Now, to share the current session customized by himself, the user  must:
* saves his map context (save plugin)
* affects good rights on it (everyone if he wants to share it with everyone)
* share the saved map context with the bounding box
For those which are not using he save plugin (as Rennes Metropole), there is no alternative to share maps.

## Acceptance criteria

- [ ] 
- [ ]

**Comment by landryb** [comment]
wrt rights, two aspects:
- a platform admin should be able to decide if anonymous users are allowed to create permalinks (at share plugin configuration level) - with that we get the old mfapp behaviour back
- an user creating a permalink should have a checkbox to decide if the permalink is public (eg viewable by anonymous users) otherwise by default only connected users can access the permalink (and for others, behaviour from #415 should apply)

@MaelREBOUX @jusabatier - do you agree with that ?

poke @fphg :)

**Comment by tdipisa** [comment]
> otherwise by default only connected users can access the permalink 

That specific point is not possible according to the MS model (a resource created by an user is visible by default to the user himself and the admins, proper permission rules need otherwise to be set for groups the user belongs to). We can try to foresee a possible solution to propose for this.

**Comment by landryb** [comment]
> > otherwise by default only connected users can access the permalink
> 
> That specific point is not possible according to the MS model (a resource created by an user is visible by default to the user himself and the admins, proper permission rules need otherwise to be set for groups the user belongs to). We can try to foresee a possible solution to propose for this.

this particular default behaviour was also discussed in #274. A platform admin might want maps/contexts to be visible by anyone by default...

**Comment by catmorales** [comment]
> A platform admin might want maps/contexts to be visible by anyone by default...

Yes I agree, it is the aim of the permalink, and one of the distinction I see with Share plugin. By default a permalink must be public (visible by everyone).

**Comment by tdipisa** [comment]
@landryb @catmorales 

I'm sorry but I think we are not understanding each other on this point. Let me better explain the meaning of my comment above:

>a platform admin should be able to decide if anonymous users are allowed to create permalinks (at share plugin configuration level) - with that we get the old mfapp behaviour back

ok

>an user creating a permalink should have a checkbox to decide if the permalink is public (eg viewable by anonymous users)

ok

> otherwise by default only connected users can access the permalink 

Is that specific point that is not possible according to the current MS model as I've explained above. If not public, a resource is visible only to the user who created it or the admins otherwise a specific permission need to be configured for groups

Therefore, if your main goal is to make the user able to decide if a permalink must public or not, it is not a problem.

**Comment by catmorales** [comment]
> Therefore, if your main goal is to make the user able to decide if a permalink must public or not, it is not a problem.

So I think it's OK. @landryb ?

**Comment by tdipisa** [comment]
@catmorales I will provide a most complete feedback later today as as result of our check today
